### PR TITLE
[4.0] [cosmetics] Remove variable type mixing in display controller / view

### PR DESCRIPTION
**Pull request by degobbis**
### Summary of Changes
~A better way to set the FTP credentials for the view~
~1. we should define the property in the view in which we want to fill it later~
~2. we should use the set method from the base class to set a value~
~3. the marking as a reference is no longer necessary~

After further research it seems that the variable $this->ftp holding the return value of the setCredentialsFromRequest() method is not used and / or directly overwritten in the associated view with the FTP credentials from the model.

### Testing Instructions
Code review

### CODE REVIEW

File: administrator/components/com_joomlaupdate/src/Controller/DisplayController.php [hunk 1]
```diff
@@ -47,7 +47,7 @@ public function display($cachable = false, $urlparams = false)
 		if ($view = $this->getView($vName, $vFormat))
 		{
 			$ftp = ClientHelper::setCredentialsFromRequest('ftp');
-			$view->ftp = &$ftp;
+			$view->set('ftp', $ftp);
```
Comment (chmst) [inline administrator/components/com_joomlaupdate/src/Controller/DisplayController.php]: maybe $view->set('ftp', ClientHelper::setCredentialsFromRequest('ftp'));

Comment (degobbis) [reply administrator/components/com_joomlaupdate/src/Controller/DisplayController.php]: thx

File: administrator/components/com_joomlaupdate/src/Controller/DisplayController.php [hunk 2]
```diff
@@ -46,9 +46,6 @@ public function display($cachable = false, $urlparams = false)
 		// Get and render the view.
 		if ($view = $this->getView($vName, $vFormat))
 		{
-			$ftp = ClientHelper::setCredentialsFromRequest('ftp');
```
Comment (Quy) [inline administrator/components/com_joomlaupdate/src/Controller/DisplayController.php:49]: Please remove `use Joomla\CMS\Client\ClientHelper;`

Comment (degobbis) [reply administrator/components/com_joomlaupdate/src/Controller/DisplayController.php:49]: Oops, the method call should not have been removed

File: administrator/components/com_joomlaupdate/src/Controller/DisplayController.php [hunk 3]
```diff
@@ -46,8 +46,7 @@ public function display($cachable = false, $urlparams = false)
...
-			$view->ftp = &$ftp;
+			ClientHelper::setCredentialsFromRequest('ftp');
```
Comment (richard67) [inline administrator/components/com_joomlaupdate/src/Controller/DisplayController.php:49]: Shouldn't it be `$view->ftp = ClientHelper::setCredentialsFromRequest('ftp');` here?

Am not sure, that's why I ask.

Comment (matrikular) [reply administrator/components/com_joomlaupdate/src/Controller/DisplayController.php:49]: As Guido wrote in the description: [...] the variable $this->ftp holding the return value of the setCredentialsFromRequest() method is not used and / or directly overwritten in the associated view with the FTP credentials from the model. [...]. 

The use of the _setCredentialsFromRequest_ method and / or it's return value is not very consistent in J!3/4. In this case, although it has not caused any harm in the past, it is not good practice either, ... cosmetics.

**Comment by chmst** [comment]
I have tested this item :white_check_mark: successfully on e57dc08af096dece4e72357cb2731361d62b989c<br /><br />Code inspect, looks good.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30043">issues.joomla.org/tracker/joomla-cms/30043</a>.</sub>

**Comment by matrikular** [comment]
I have tested this item :white_check_mark: successfully on e57dc08af096dece4e72357cb2731361d62b989c<br /><br />Looks good on code review.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30043">issues.joomla.org/tracker/joomla-cms/30043</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30043">issues.joomla.org/tracker/joomla-cms/30043</a>.</sub>

**Comment by wilsonge** [comment]
I want a practical test of this too please as it covers Joomla Update. Code review is good enough for one test but not both

**Comment by richard67** [comment]
> I want a practical test of this too please as it covers Joomla Update. Code review is good enough for one test but not both

@degobbis Please provide testing instructions for a practical test. Thanks in advance.

**Comment by degobbis** [comment]
@richard67 @wilsonge 
For testing we need an older NightlyBuild to exerzise an Joomla update, where I can get one?

**Comment by degobbis** [comment]
While playing through the test instruction I noticed that a test is very complex and not easy to do.

Since the FTP credentials were implemented as a workaround for the wwwrun problem on web servers, such a server is necessary for a reliable test.

Already the installation of Joomla on such a server is the first hurdle.

At least two installations are necessary, because resetting the installation after the update is not so easy and could falsify the result because of the file permissions.

Do we have professionals who could test this?

**Comment by richard67** [comment]
There is another PR dealing with Joompaupdate Upload & Update and FTP: #28029 . Maybe the author, @twister65 can help with testing here? Or maybe @Quy can do?

**Comment by wilsonge** [comment]
OK Doesn't look like we'll get testers :/ Thanks

**Pull request merged**